src/lightning_app/utilities/proxies.py

```diff
@@ -85,15 +92,13 @@ def __post_init__(self):
         self.work_state = None
 
     def __call__(self, *args, **kwargs):
-        provided_none = len(args) == 1 and args[0] is None
-        use_args = len(kwargs) > 0 or (len(args) > 0 and not provided_none)
-
         self._validate_call_args(args, kwargs)
         args, kwargs = self._process_call_args(args, kwargs)
 
         call_hash = self.work._call_hash(self.work_run, args, kwargs)
-        entered = call_hash in self.work._calls
-        returned = entered and "ret" in self.work._calls[call_hash]
+        succeeded = call_hash in self.work._calls[CacheCallsKeys.SUCCEEDED_CALL_HASHES]
+        entered = call_hash in self.work._calls or succeeded
```

Comment:
`self.work._calls[CacheCallsKeys.SUCCEEDED_CALL_HASHES]` should have a constant memory size. 

We want to answer the question that if we have seen this `call_hash` hash before or not, here we can use a bloom filter.

Reply:
This is growing as new keys can be added.

Reply:
Why it should be growing ? the memory footprint of a running a lightning app should be constant. no ?

Reply:
I changed slightly the architecture. But the state is constant if nothing modifies it or no work is run with new arguments.